tests/pyspark_delta/test_pyspark_delta.py

```diff
@@ -1,52 +1,61 @@
-        "org.apache.spark.sql.delta.catalog.DeltaCatalog",
-    )
-)
-
-spark = configure_spark_with_delta_pip(builder).getOrCreate()
-
-def test_reference_table_1_content():
-    versions = [0, 1]
-    for v in versions:
-        actual_df = spark.read.format("delta").option("versionAsOf", v).load("out/tables/generated/reference_table_1/delta")
-        expected_df = spark.read.format("parquet").load(f"out/tables/generated/reference_table_1/expected/v{v}/table_content.parquet")
-        chispa.assert_df_equality(actual_df, expected_df)
-
-
-def test_reference_table_2_content():
-    versions = [0, 1]
-    for v in versions:
-        actual_df = spark.read.format("delta").option("versionAsOf", v).load("out/tables/generated/reference_table_2/delta")
-        expected_df = spark.read.format("parquet").load(f"out/tables/generated/reference_table_2/expected/v{v}/table_content.parquet")
-        chispa.assert_df_equality(actual_df, expected_df)
-
-
-def test_reference_table_3_content():
-    versions = [0, 1, 2]
-    for v in versions:
-        actual_df = spark.read.format("delta").option("versionAsOf", v).load("out/tables/generated/reference_table_3/delta")
-        expected_df = spark.read.format("parquet").load(f"out/tables/generated/reference_table_3/expected/v{v}/table_content.parquet")
-        chispa.assert_df_equality(actual_df, expected_df)
-
-
-def test_all_readers():
-    test_config = [
-        {"rt": 1, "versions": [0, 1]},
-        {"rt": 2, "versions": [0, 1]},
-        {"rt": 3, "versions": [0, 1, 2]},
-    ]
-    for config in test_config:
-        for v in config["versions"]:  
-            rt_num = config["rt"]
-            actual_df = spark.read.format("delta").option("versionAsOf", v).load(f"out/tables/generated/reference_table_{rt_num}/delta")
-            expected_df = spark.read.format("parquet").load(f"out/tables/generated/reference_table_{rt_num}/expected/v{v}/table_content.parquet")
-            chispa.assert_df_equality(actual_df, expected_df)
+import pytest
+
+TEST_ROOT = Path('../out/reader_tests/')
+
+
+class ReadCase(NamedTuple):
+    delta_root: Path
+    version: Optional[int]
+    parquet_root: Path
+    name: str
+    description: str
+    min_reader_version: int
+    min_writer_version: int
+
+
+cases: List[ReadCase] = []
+
+for path in (TEST_ROOT / 'generated').iterdir():
+    if path.is_dir():
+        with open(path / 'table_metadata.json') as f:
+            case_metadata = json.loads(f)
+
+        for version_path in (path / 'expected').iterdir():
+            if version_path.is_dir():
+                if version_path.name[0] == 'v':
+                    version = int(version_path.name[1:])
+                elif version_path.name == 'latest':
+                    version = None
+                else:
+                    continue
+
+                with open(version_path / 'expected_metadata.json') as f:
+                    expected_metadata = json.loads(f)
+
+                case = ReadCase(
```

Comment:
Is this variable used? Running the linter should identify this unused variable. Maybe it's missing a cases.append(case)?

Reply:
I did run `make lint` and it didn't catch this. But yes, I needed to add `cases.append(case)` :)